### PR TITLE
Feature 2501 pds solutions: add scc

### LINKED ISSUE #2501: PDS solutions: Add `scc` as an alternative to `cloc`
## Situation
- while parsing big files, cloc eats up much memory
- cloc ignores big files (which makes the results not 100% reliable)

## Wanted
Add `scc` (https://github.com/boyter/scc/ / MIT license) to the "cloc" analytics container as a (hopefully) better alternative.
The output shall be Json compatible to the orignial cloc's output.

## ToDo
- rename `cloc` in sechub-pds-solutions/ directory to `loc` to avoid confusion
- add `scc` to the pds-solution
- verify if the output works with the current analytics processing (run a test + check statistics database tables)

### PR DESCRIPTION
**Pull request by sven-dmlr**
- closes #2501

### CODE REVIEW

File: sechub-pds-solutions/loc/README.adoc [hunk 1]
```diff
@@ -3,13 +3,12 @@
 :toc:
 :numbered:
 
-= cloc + PDS
+= loc (lines of code) analytics tools + PDS
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/README.adoc:6]:
```suggestion
= LoC analytics tools + PDS

Lines of Code (LoC) analytics tools measure the number of lines of code in source code files.
```

File: sechub-pds-solutions/loc/docker/loc-Debian.dockerfile [hunk 2]
```diff
@@ -8,13 +8,16 @@ FROM ${BASE_IMAGE}
 # See: https://ryandaniels.ca/blog/docker-dockerfile-arg-from-arg-trouble/
 
 LABEL org.opencontainers.image.source="https://github.com/mercedes-benz/sechub"
-LABEL org.opencontainers.image.title="SecHub cloc+PDS Image"
-LABEL org.opencontainers.image.description="A container which combines cloc with the SecHub Product Delegation Server (PDS)"
+LABEL org.opencontainers.image.title="SecHub loc+PDS Image"
+LABEL org.opencontainers.image.description="A container which combines loc analytics tools with the SecHub Product Delegation Server (PDS)"
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/docker/loc-Debian.dockerfile:12]:
```suggestion
LABEL org.opencontainers.image.description="A container which combines lines of code (LoC) counting analytics tools with the SecHub Product Delegation Server (PDS)"
```

File: sechub-pds-solutions/loc/docker/loc-Debian.dockerfile [hunk 3]
```diff
@@ -27,9 +30,17 @@ COPY pds-config.json "$PDS_FOLDER"/pds-config.json
 RUN export DEBIAN_FRONTEND=noninteractive && \
     apt-get update && \
     apt-get --assume-yes upgrade && \
-    apt-get --assume-yes install wget perl && \
+    apt-get --assume-yes install perl wget yq && \
     apt-get --assume-yes clean
 
+# Install scc
+RUN cd "$DOWNLOAD_FOLDER" && \
+    wget --no-verbose "https://github.com/boyter/scc/releases/download/v${SCC_VERSION}/scc_${SCC_VERSION}_Linux_x86_64.tar.gz" && \
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/docker/loc-Debian.dockerfile:38]: Please use the checksums.txt file to cross-check the integrity of the downloaded file: https://github.com/boyter/scc/releases/download/v3.1.0/checksums.txt. Source: https://github.com/boyter/scc/releases/.

File: sechub-pds-solutions/loc/docker/scripts/scc.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    extracted_folder="$PDS_JOB_EXTRACTED_SOURCES_FOLDER"
+else
+    echo ""
+    echo "ERROR: No source code."
+    exit 1
+fi
+
+# Call scc and convert Yaml to Json
+# We need to turn the "version" attribute to "cloc_version" so the SecHub server will parse it.
+scc -f cloc-yaml "$extracted_folder" > scc.yaml && \
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/docker/scripts/scc.sh:19]: I would prefer the full form of the `-f` parameter. Easier to read and maintain.

File: sechub-pds-solutions/loc/docker/scripts/scc.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    echo "Found extracted sources"
+    extracted_folder="$PDS_JOB_EXTRACTED_SOURCES_FOLDER"
+else
+    echo ""
+    echo "ERROR: No source code."
+    exit 1
+fi
+
+# Call scc and convert Yaml to Json
+# We need to turn the "version" attribute to "cloc_version" so the SecHub server will parse it.
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/docker/scripts/scc.sh:18]: Please create an issue on GitHub for this. Maybe, the parser for the loc component can use another field/combination to recognize the report format.

File: sechub-pds-solutions/loc/docker/scripts/scc_mock.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
+
+echo "Running PDS Scc Mock"
+cp "$MOCK_FOLDER/scc-mock.json" "$PDS_JOB_RESULT_FILE"
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/docker/scripts/scc_mock.sh:5]: Please add as a comment, what repository you used to generate the mock file output. We might have to adjust that if `scc` changes in the future (have a look at the `cloc` mock file for an example.

File: sechub-pds-solutions/loc/helm/pds-loc/Chart.yaml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 # SPDX-License-Identifier: MIT
 
 apiVersion: v2
-name: pds-cloc
-description: SecHub PDS + cloc as Helm chart for Kubernetes
+name: pds-loc
+description: SecHub PDS + loc analytics tools as Helm chart for Kubernetes
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/helm/pds-loc/Chart.yaml:5]:
```suggestion
description: SecHub PDS + loc (lines of code) analytics tools as Helm chart for Kubernetes
```

File: sechub-pds-solutions/loc/helm/pds-loc/README.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<!-- SPDX-License-Identifier: MIT --->
+# loc + PDS
+
+This Helm chart enables one to deploy source code analyzer tools (cloc, scc) and the [SecHub Product Delegation Server (PDS)](https://mercedes-benz.github.io/sechub/latest/sechub-product-delegation-server.html) into a Kubernetes environment. It is recommended to use loc + PDS together with [SecHub](https://mercedes-benz.github.io/sechub/).
```
Comment (Jeeppler) [inline sechub-pds-solutions/loc/helm/pds-loc/README.md:4]:
```suggestion
This Helm chart enables one to deploy lines of code analyzer tools (cloc, scc) and the [SecHub Product Delegation Server (PDS)](https://mercedes-benz.github.io/sechub/latest/sechub-product-delegation-server.html) into a Kubernetes environment. It is recommended to use loc + PDS together with [SecHub](https://mercedes-benz.github.io/sechub/).
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Overall, good job 👍 
However, I found a few issues.

File: sechub-pds-solutions/loc/docker/loc-Debian.dockerfile (hunk 2 above)
Comment (sven-dmlr) [reply sechub-pds-solutions/loc/docker/loc-Debian.dockerfile:12]: ok. changed

File: sechub-pds-solutions/loc/docker/scripts/scc_mock.sh (hunk 6 above)
Comment (sven-dmlr) [reply sechub-pds-solutions/loc/docker/scripts/scc_mock.sh:5]: ok. done

File: sechub-pds-solutions/loc/helm/pds-loc/Chart.yaml (hunk 7 above)
Comment (sven-dmlr) [reply sechub-pds-solutions/loc/helm/pds-loc/Chart.yaml:5]: ok. changed

**Pull request merged**